snapcraft/internal/lxd.py

```diff
@@ -29,7 +29,7 @@
 import yaml
 
 from snapcraft.internal.errors import SnapcraftEnvironmentError
-from snapcraft.internal import common
+from snapcraft.internal.common import format_snap_name, is_snap
```

Comment:
No, scratch that. I see now what you mean. I am more or less ok with your approach.

However, I always avoid importing symbols, so I never patch like this. It is a little confusing that the behaviour of is_snap will be patched in the test only on the call from the lxd module. The fact that you need two different behaviours for that method probably means that you are not testing a unit of code. I am not strongly against this, mainly because the lack of testability is not your fault, and adding this test helps a little to improve things later. But it's a bad smell that we have to change the import to use a test double.